pkg/infra/error.go

```diff
@@ -0,0 +1,17 @@
+package infra
```

Comment:
it's actually not very clear to me why we need to define these errors?

Reply:
Making const and type of the errors.
For ex, any kind of connection error will return as `ConnectError` and the details of reason/stack in log.

Reply:
i don't think this is really necessary - they are merely common strings to save some hand-tying, instead of actual errors, which makes them less useful.

Reply:
ok